parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/HadoopStreams.java

```diff
@@ -51,7 +52,7 @@ public class HadoopStreams {
   public static SeekableInputStream wrap(FSDataInputStream stream) {
     Objects.requireNonNull(stream, "Cannot wrap a null input stream");
     if (byteBufferReadableClass != null && h2SeekableConstructor != null &&
-        byteBufferReadableClass.isInstance(stream.getWrappedStream())) {
+        isWrappedStreamByteBufferReadable(stream)) {
```

Comment:
this is really going into the internals of the hadoop classes and potentially tricky if there is any dynamic decision making in the inner class. The good news there is I don't see anything doing that.

there is a way to ask (hadoop 3.2+) if a stream does support the API before calling, using the StreamCapabilities interface.
https://issues.apache.org/jira/browse/HDFS-14111

```java
if (stream.hasCapability( "in:readbytebuffer") {
  // stream is confident it has the api
) else {
  // do the checking of the inner class
}